presto-main/src/main/java/com/facebook/presto/sql/planner/SplitSourceFactory.java

```diff
@@ -135,6 +137,17 @@ private Visitor(Session session, StageExecutionDescriptor stageExecutionDescript
             return ImmutableMap.of(node.getId(), splitSource);
         }
 
+        private SplitSchedulingStrategy getSplitSchedulingStrategy(StageExecutionDescriptor stageExecutionDescriptor, PlanNodeId scanNodeId)
+        {
+            if (stageExecutionDescriptor.isRecoverableGroupedExecution()) {
```

Comment:
nit: Can we use switch over `stageExecutionDescriptor.getStageExecutionStrategy` ?

Reply:
No because `stageExecutionDescriptor::isScanGroupedExecution` needs to take `PlanNodeId`.